analyzers/tests/SonarAnalyzer.UnitTest/TestCases/MemberInitializedToDefault.CSharp11.cs

```diff
@@ -1,32 +1,75 @@
-    public UIntPtr field7 = UIntPtr.Zero;           // Noncompliant
-    public IntPtr field8 = new IntPtr(0);           // Noncompliant
+    public IntPtr field5 = IntPtr.Zero; // Noncompliant
+    public IntPtr field6 = 0x0000000000000000; // Noncompliant
+    public UIntPtr field7 = UIntPtr.Zero; // Noncompliant
+    public IntPtr field8 = new IntPtr(0); // Noncompliant
     public IntPtr field9 = new IntPtr(staticField); // Compliant
 
-    public UIntPtr field10 = new UIntPtr(0);        // Noncompliant
-    public UIntPtr field11 = new();                 // Noncompliant
-    public UIntPtr field12 = new UIntPtr { };       // Noncompliant
+    public UIntPtr field10 = new UIntPtr(0); // Noncompliant
+    public UIntPtr field11 = new();  // Noncompliant
+    public UIntPtr field12 = new UIntPtr { }; // Noncompliant
 
-    public IntPtr Property1 { get; set; } = 0;      // Noncompliant
-    public IntPtr Property2 { get; set; } = 42;     // Compliant
+    public IntPtr Property1 { get; set; } = 0; // Noncompliant
+    public IntPtr Property2 { get; set; } = 42; // Compliant
 
-    public UIntPtr Property3 { get; init; } = 0;    // Noncompliant
-    public UIntPtr Property4 { get; init; } = 42;   // Compliant
+    public UIntPtr Property3 { get; init; } = 0; // Noncompliant
+    public UIntPtr Property4 { get; init; } = 42; // Compliant
 
     public IntPtr Property5 { get; set; } = 0 * 20 - 0; // FN - Expression is not evaluated
     public UIntPtr Property6 { get; set; } = 0 * 20 - 0; // FN - Expression is not evaluated
 }
+
+public readonly struct FooStruct
+{
+    public FooStruct() { }
+
+    public int Value { get; init; } // Compliant
+    public bool BoolValue { get; init; } // Compliant
+
+    public int ValueDefault { get; init; } = 0; // Noncompliant
+    public bool BoolValueDefault { get; init; } = false; // Noncompliant
+}
+
+public struct BarStruct
+{
+    public int someField = 0; // Noncompliant FP for C# 11 due to the auto-default-struct C# 11 feature https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-11#auto-default-struct
+    public BarStruct(int dummy) { }
+}
+
+public struct FooBarStruct
+{
+    public int someField; // Compliant - does not raise a CS0171 issue anymore due to the auto-default-struct C# 11 feature https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/csharp-11#auto-default-struct
+    public required int someRequiredField; // Compliant - does not raise a CS0171 and the compiler will default this to 0.
+
+    public FooBarStruct(int dummy) { }
+}
+
+public class ClassWithRequiredProperties
+{
+    public required int RequiredProperty { get; init; }
+    public int AnotherProperty { get; set; }
+
+    public ClassWithRequiredProperties()
+    {
+        RequiredProperty = 0; // FN - the compiler will set this automatically to 0
+    }
+
+    [SetsRequiredMembers]
+    public ClassWithRequiredProperties(int prop)
+    {
+        RequiredProperty = prop;
```

Comment:
it's interesting that even if I don't set the required property from the constructor at all it is still defaulted to 0.

I did not add a struct example as it's the caller that needs to always initialize the field through the constructor.